### PR TITLE
Improve prebuilt baselining: warn on error by default, save all project.assets.json

### PR DESCRIPTION
**Pull request by dagood**
A few improvements to prebuilt baselining: #770, #774.

### CODE REVIEW

File: netci.groovy [hunk 1]
```diff
@@ -13,7 +13,8 @@ def addArchival(def job) {
   // Grab these logs from all of those locations.
   [ "", "source-build/", "tarball-output/"].each { logRoot ->
     archivalSettings.addFiles("${logRoot}bin/logs/*")
-    archivalSettings.addFiles("${logRoot}bin/prebuilt-report/*")
+    archivalSettings.addFiles("${logRoot}bin/prebuilt-report/**/*")
```
Comment (crummel) [inline netci.groovy]: Does this still pick up loose files in `prebuilt-report`?  The XML summaries will still be going in there, right?

Comment (dagood) [reply netci.groovy]: It should--`**` is meant to match no/current directory as well as all recursive subdirectories.

Comment (dagood) [reply netci.groovy]: Testing this with https://ci.dot.net/job/dotnet_source-build/job/release_2.1/job/GenPRTest/job/Ubuntu16.04_Release_prtest/1/

**Comment by dagood** [comment]
@dotnet-bot test RHEL7.2 Tarball Release
No idea what caused [this](https://ci.dot.net/job/dotnet_source-build/job/release_2.1/job/RHEL7.2_Tarball_Release_prtest/39/), retrying in case it's flaky. Would be nice to have https://github.com/Microsoft/msbuild/issues/1038. `error MSB4166: Child node "3" exited prematurely.`

**Comment by crummel** [comment]
> This location can be changed by setting the MSBUILDDEBUGPATH environment variable to a different directory.

Wonder if it's worth adding this variable in [repos/dir.props](https://github.com/dotnet/source-build/blob/89a692e199d525baaef1be0d46a0366787e95561/repos/dir.props#L45)?

**Comment by dagood** [comment]
I'll add that to this PR, definitely worth it if it works.

**Pull request merged**